### ISSUE TITLE
[🐛 Bug]: Failed Vercel build doesn't produce a non-zero result

### ISSUE THREAD

**Issue by dario-piotrowicz** [cloudflare/next-on-pages#210]
If the `--watch` flag is not used and the Vercel build fails the whole process should exit with a non-zero status (so that potential callers can see that the command failed), this is currently not happening since we sort of just ignore a failed build:
https://github.com/cloudflare/next-on-pages/blob/d7e62c7fb99241565b41a664abfde29bbac99755/src/buildApplication/buildApplication.ts#L48

This should be improved so that a non-zero status is returned instead (meaning that a failed build, without using the `--watch` flag should call `process.exit(X)`).

___

Basically if I have the following nodejs code:
```js
    const { status } = spawnSync("npx", ["@cloudflare/next-on-pages"], {
      cwd: join(appDir),
      stdio: "inherit",
    });
```
and the application in `appDir` causes the Vercel build to error, then `status` should be non-zero (it currently is `0` and we get a non-zero value only if the process fails for different reasons)
